### ISSUE TITLE
Add documentation in doc/, in particular for the known Gen 1 bugs

### ISSUE THREAD

**Issue by Rangi42** [pret/pokered#258]
pokecrystal has at least 66 [known bugs](https://github.com/pret/pokecrystal/blob/master/docs/bugs_and_glitches.md), as well as various [design flaws](https://github.com/pret/pokecrystal/blob/master/docs/design_flaws.md) (not really bugs, but still things a hack project would want to correct). pokegold has at least three [unique bugs](https://github.com/pret/pokegold/issues/52). I expect that pokered has many more:

- https://glitchcity.info/wiki/List_of_natural_glitches_in_Generation_I
- https://glitchcity.info/wiki/Category:Generation_I_glitches
- https://bulbapedia.bulbagarden.net/wiki/List_of_glitches_in_Generation_I

Ideally these would be documented, with minimal fixes if possible.

Design flaws might not be worth documenting (since they're subjective and Gen 1 arguably has enough of them that the appropriate fix is to switch to Gen 2 or port its subsystems *en masse*—see [pokered-crysaudio](https://github.com/dannye/pokered-crysaudio), [pokered-gbc](https://github.com/dannye/pokered-gbc), [pokered-clean](https://github.com/TheFakeMateo/pokered-clean), etc), but definite bugs are.

**Comment by Rangi42** [comment]
With glitchcity.info [shutting down](https://forums.glitchcity.info/index.php?topic=9114.0), it would be that much more useful to have a central lookup for all Gen 1 bugs, with source code references.

**Comment by mid-kid** [comment]
I feel like the pret docs focus more on inappropriate behavior of the code, while glitchcity.info focusses more on practical applications of the bugs. These are two very different approaches to the documentation that are hard to combine while keeping consistent.

**Comment by Rangi42** [comment]
You're right. I don't mean to document every single glitch they do. But their "natural glitch" list is a good starting point for a level of bug documentation similar to pokecrystal, and that first link to the forums already has YouTube and pokered links.

**Comment by Rangi42** [comment]
There are also some bugs GCL wouldn't have documented which are nevertheless bugs in the code; for example:

```
; function to check the tile ahead to determine if the character should get on land or keep surfing
; sets carry if there is a collision and clears carry otherwise
; It seems that this function has a bug in it, but due to luck, it doesn't
; show up. After detecting a sprite collision, it jumps to the code that
; checks if the next tile is passable instead of just directly jumping to the
; "collision detected" code. However, it doesn't store the next tile in c,
; so the old value of c is used. 2429 is always called before this function,
; and 2429 always sets c to 0xF0. There is no 0xF0 background tile, so it
; is considered impassable and it is detected as a collision.
CollisionCheckOnWater::
```

**Comment by mid-kid** [comment]
Oh, yeah, all I'm saying is that this wiki is probably not a good replacement for what glitchcity.info used to provide.

**Comment by Rangi42** [comment]
GCL is shutting down in a week (https://forums.glitchcity.info/index.php?topic=9149). Of course their documented glitches will still be archived, but this would be a good time to start a Bugs and Glitches doc for pokered.

**Comment by SatoMew** [comment]
Here's [the current link](https://glitchcity.wiki/) to the GCRI wiki. The contents were carried over from the old GCL wiki.

I'll be sharing some info on bugs in a follow-up comment.

**Comment by vulcandth** [comment]
When I get some time; i'll see about starting this. I'll create a docs/bugs_and_glitches.md formatted similarly to pokecrystal's. I'll add a bug fix or two... and create a PR to merge. The idea, is that this will serve as a template for others to add to.

**Comment by Rangi42** [comment]
It can start as a wiki page for easy contribution; I thought someone had already made one in the pokered wiki, but apparently not.

**Comment by vulcandth** [comment]
> It can start as a wiki page for easy contribution; I thought someone had already made one in the pokered wiki, but apparently not.

Yeah; that might be a better idea. Then we can look at merging it into the codebase later. I still do like the idea of it eventually making it into the codebase. As things in the codebase tend to be "vetted", and gives downstream users *simple* bug fixes they can implement with little worry of causing further bugs (in the vanilla rom).

**Comment by coltongit** [comment]
The 5 Pokéballs text issue and the Cinnabar Island glitch have both been given wiki articles. The latter just needs expansion to deal with the other 2x2 block encounter glitches.

**Comment by Rangi42** [comment]
...please don't start adding issue comments with notifications for every incremental update to the wiki. There are possibly hundreds of Gen 1 bugs to document, certainly more than Gen 2. Just, go ahead and add to the wiki whenever you like, other people can too, and eventually if it looks like a comprehensive document of bugs we can consider putting it in the repo.

**Comment by coltongit** [comment]
> > It can start as a wiki page for easy contribution; I thought someone had already made one in the pokered wiki, but apparently not.
> 
> Yeah; that might be a better idea. Then we can look at merging it into the codebase later. I still do like the idea of it eventually making it into the codebase. As things in the codebase tend to be "vetted", and gives downstream users _simple_ bug fixes they can implement with little worry of causing further bugs (in the vanilla rom).

Added the wiki page for the bugs and glitches document, still needs video docs otherwise it's pretty well formatted the same as pokegold's and pokecrystal's document: https://github.com/pret/pokered/wiki/Bugs-and-Glitches

**Comment by halian** [comment]
> Added the wiki page for the bugs and glitches document, still needs video docs otherwise it's pretty well formatted the same as pokegold's and pokecrystal's document: https://github.com/pret/pokered/wiki/Bugs-and-Glitches

@rawr51919 unfortunately, it's been deleted :(
